agent/consul/operator_raft_endpoint.go

```diff
@@ -2,15 +2,43 @@ package consul
 
 import (
 	"fmt"
+
 	"net"
 
+	"github.com/hashicorp/consul/acl"
+
 	"github.com/hashicorp/raft"
 	"github.com/hashicorp/serf/serf"
 
 	"github.com/hashicorp/consul/agent/metadata"
 	"github.com/hashicorp/consul/agent/structs"
 )
 
+// RaftLeaderTransfer is used to attempt a leadership transfer
+func (op *Operator) RaftLeaderTransfer(args *structs.LeaderTransferRequest, reply *structs.LeadershipTransferResponse) error {
+	if done, err := op.srv.ForwardRPC("Operator.RaftLeaderTransfer", args, reply); done {
+		return err
+	}
+	// This action requires operator read access.
+	authz, err := op.srv.ResolveToken(args.Token)
+	if err != nil {
+		return err
+	}
+	if err := op.srv.validateEnterpriseToken(authz.Identity()); err != nil {
+		return err
+	}
+	if authz.OperatorRead(nil) != acl.Allow {
```

Comment:
Should this be `OperatorWrite`? Or `OperatorRead`?

Reply:
yes definitely!

Reply:
I will also make sure that the ACL test use a token with the right access rights to catch this